unittest/test_transformer.py

```diff
@@ -108,3 +109,120 @@ def test_equivalent_pipeline():
     assert not transformer._transformer.skip_equivalent
     assert not transformer._transformer.projections_equivalent
     assert not transformer._transformer.projections_exact_same
+
+
+def test_4d_transform():
+    transformer = Transformer.from_pipeline("+init=ITRF2008:ITRF2000")
+    assert_almost_equal(
+        transformer.transform(
+            xx=3513638.19380, yy=778956.45250, zz=5248216.46900, tt=2008.75
+        ),
+        (3513638.1999428216, 778956.4532640711, 5248216.453456361, 2008.75),
+    )
+
+
+def test_2d_with_time_transform():
+    transformer = Transformer.from_pipeline("+init=ITRF2008:ITRF2000")
+    assert_almost_equal(
+        transformer.transform(xx=3513638.19380, yy=778956.45250, tt=2008.75),
+        (3513638.1999428216, 778956.4532640711, 2008.75),
+    )
+
+
+def test_4d_transform_crs_obs1():
+    transformer = Transformer.from_proj(7789, 8401)
+    assert_almost_equal(
+        transformer.transform(
+            xx=3496737.2679, yy=743254.4507, zz=5264462.9620, tt=2019.0
+        ),
+        (3496737.2679, 743254.4507, 5264462.962, 2019.0),
```

Comment:
This doesn't seem right. From PROJ master:

```
$ echo 3496737.2679 743254.4507 5264462.9620 2019.0 |./cs2cs EPSG:7789 EPSG:8401
3496737.76	743253.99 5264462.70 2019.0
```

I don't know what tolerance is used in `assert_almost_equal` but a difference of ~70 cm in coordinates is significant and should, in my opinion, not be deemed equal.

Just for a bit of perspective: Often time these 4D transforms are used to account for the effect of continents moving over time. The Eurasian plate moves at a rate of 2-2.5 cm/year.